### ISSUE TITLE
Creation of new RAM role on another account is not possible when using assume_role

### ISSUE THREAD

**Issue by amwk** [terraform-providers/terraform-provider-alicloud#1534]
Hi there,

I have a problem creating new RAM role on another account when using assume_role. This works fine only when Access/Secret Keys are provided, but not with STSToken. Result is creation of new RAM role on current account, not the target one. 

### Terraform Version
```
$ terraform -v
Terraform v0.12.6
+ provider.alicloud v1.53.0
```

### Affected Resource(s)
- alicloud_ram_role

### Terraform Configuration Files
```hcl
provider "alicloud" {
  ecs_role_name = "Root-terraform-role"
  region     = "us-east-1"

  assume_role {
    role_arn           = "acs:ram::<hidden>:role/Root-Account-role"
    policy             = "POLICY"
    session_name       = "SESSION_NAME"
    session_expiration = 999
  }
}

data "alicloud_account" "current"{
}

resource "alicloud_ram_role" "ro-role" {
  name = "TEST_ROLE"

  document = <<EOF
  {
    "Statement": [
        {
            "Action": "sts:AssumeRole",
            "Condition": {
                "StringEquals": {
                    "saml:recipient": "https://signin.alibabacloud.com/saml-role/sso"
                }
            },
            "Effect": "Allow",
            "Principal": {
                "Federated": [
                    "acs:ram::${data.alicloud_account.current.id}:saml-provider/<hidden>"
                ]
            }
        }
    ],
    "Version": "1"
  }
  EOF
  description = "SSO-Federation ReadOnly"
  force = true
}

```

### Debug Output

https://gist.github.com/amwk/6ce10d4b39e6c7199263231c5e6c8b79

### Expected Behavior
New RAM role is created in target account. 

### Actual Behavior
New RAM role is created in current account. 

### Steps to Reproduce
1. You have to have two accounts to reproduce this problem. 
2. Use terrafrom configuration provided above
3. terraform init
4. terraform apply

### Important Factoids
This actually works, but only when you use Access/Secret key instead of STSToken from assume role

**Comment by xiaozhu36** [comment]
HI @amwk This issue has been fixed by 1.55.4 and please check it. Please let me know if you still have any issues.

**Comment by xiaozhu36** [comment]
Closed by no reply.
